ros2topic/ros2topic/verb/pub.py

```diff
@@ -80,6 +80,9 @@ def add_arguments(self, parser, cli_name):
         group.add_argument(
             '-1', '--once', action='store_true',
             help='Publish one message and exit')
+        group.add_argument(
+            '-m', '--wait-matching-subscriptions', type=nonnegative_int, default=1,
```

Comment:
I'll still argue that having the default be `1` is a substantial difference in behavior from earlier versions of ROS 2 and ROS 1.  As long as we are OK with that difference in behavior, then we can go forward, but I also think we then need to have a release note about it.

Reply:
> As long as we are OK with that difference in behavior, then we can go forward, but I also think we then need to have a release note about it.

I can bring this up in the next ROS 2 meeting and ask for opinions, mentioning this in the release notes sounds fine to me.

> I'll still argue that having the default be 1 is a substantial difference in behavior from earlier versions of ROS 2 and ROS 1

This wasn't a problem in ROS 1 as it has an unique graph state that all nodes are aware.
When you created a node and a publisher, it was guaranteed that you were going to see all already existing subscriptions.

This is not the case in ROS 2, and I have got confused with `ros2 topic pub -1 --qos-reliability reliable <topic_name> <topic_type> <message>` not working enough times to think the default behavior isn't intuitive.

Ideally, we would also have https://github.com/ros2/rmw/issues/295 and call it before ending the process, specially when using `--times/--once`.